tools/azsdk-cli/Azure.Sdk.Tools.Cli/Tools/AzurePipelines/PipelineDetailsTool.cs

```diff
@@ -0,0 +1,101 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+using System.CommandLine;
+using System.CommandLine.Invocation;
+using System.ComponentModel;
+using Azure.Core;
+using Azure.Sdk.Tools.Cli.Commands;
+using Azure.Sdk.Tools.Cli.Contract;
+using Azure.Sdk.Tools.Cli.Services;
+using Microsoft.TeamFoundation.Build.WebApi;
+using Microsoft.VisualStudio.Services.OAuth;
+using Microsoft.VisualStudio.Services.WebApi;
+using ModelContextProtocol.Server;
+
+namespace Azure.Sdk.Tools.Cli.Tools;
+
+[McpServerToolType, Description("Fetches data from Azure Pipelines")]
+public class PipelineDetailsTool : MCPTool
+{
+    private BuildHttpClient buildClient;
+    private readonly bool initialized = false;
+
+    private IAzureService azureService;
+    private IOutputService output;
+    private ILogger<PipelineDetailsTool> logger;
+
+    // Options
+    private readonly Option<int> buildIdOpt = new(["--build-id", "-b"], "Pipeline/Build ID") { IsRequired = true };
+    private readonly Option<string> projectOpt = new(["--project", "-p"], "Pipeline project name");
+
+    public PipelineDetailsTool(
+        IAzureService azureService,
+        IOutputService output,
+        ILogger<PipelineDetailsTool> logger
+    ) : base()
+    {
+        this.azureService = azureService;
+        this.output = output;
+        this.logger = logger;
+
+        CommandHierarchy =
+        [
+            CommandVerbDescriptions.AzurePipelines
+        ];
+    }
+
+    public override Command GetCommand()
+    {
+        var pipelineRunCommand = new Command("pipeline", "Get details for a pipeline run") { buildIdOpt, projectOpt };
+        pipelineRunCommand.SetHandler(async ctx => { await HandleCommand(ctx, ctx.GetCancellationToken()); });
+
+        return pipelineRunCommand;
+    }
+
+    public override async Task HandleCommand(InvocationContext ctx, CancellationToken ct)
+    {
+        Initialize();
+
+        var cmd = ctx.ParseResult.CommandResult.Command.Name;
+        var buildId = ctx.ParseResult.GetValueForOption(buildIdOpt);
+        var project = ctx.ParseResult.GetValueForOption(projectOpt);
+
+        logger.LogInformation("Getting pipeline run {buildId}...", buildId);
+        var result = await GetPipelineRun(project, buildId);
+        ctx.ExitCode = ExitCode;
+        output.Output(result);
+    }
+
+    private void Initialize()
+    {
+        if (initialized)
```

Comment:
I'll probably add a parent class to de-dup initialize as we add more commands, but this is fine for now.